### PR TITLE
Mkdocs based documentation static site generation

### PR DESCRIPTION
**Pull request by pkdash**
This PR consists of changes to build and deploy a static site based on Mkdocs. The documentation site will be rebuild and deployed when there is a commit to the develop branch using github action.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+site_name: CZ Net Data Best Practices
+repo_name: cznethub/data-best-practices
+repo_url: https://github.com/cznethub/data-best-practices
+edit_uri: blob/develop/docs/
+
+theme: readthedocs
```
Comment (devincowan) [inline mkdocs.yml]: The only thing I don't like about `readthedocs` theme is that I find the search very clunky

Try and compare with the material theme search:
https://cuahsi.github.io/mkdocs-example/sample_data/registering_samples/

It has search suggestions and highlighting
https://github.com/CUAHSI/mkdocs-example/blob/master/mkdocs.yml#L3-L7

**Review by devincowan — APPROVED** [review]
Looks great, thanks @pkdash!

**Review by horsburgh — APPROVED** [review]
Based on our conversation and changes made today, I think this is good to go.

**Comment by pkdash** [comment]
@horsburgh Let me know if we should use the material theme as suggested by @devincowan. You can see the use of that theme here - https://cuahsi.github.io/mkdocs-example/sample_data/registering_samples/ 

The search functionality seems to be better in the material theme.

**Comment by devincowan** [comment]
I didn't mean to take use back to debating themes, I think we already made that decision. I was just hoping we might be able to improve the search within the `readthedocs` theme.

However, it looks like that isn't an option.
Here are two explanations of how the themes conduct their search:

### Material
* https://squidfunk.github.io/mkdocs-material/blog/2021/09/13/search-better-faster-smaller/
* Client side, no page refresh
* Has search previews and highlighting 
* Includes search as you type

### Readthedocs
* https://blog.readthedocs.com/search-improvements/
* uses Elasticsearch
* No "Search as you type"
* No search previews


Probably the `readthedocs` search is perfectly fine for our purposes, I just have a strong personal preference for the `material` search.

**Comment by horsburgh** [comment]
@pkdash and @devincowan - on today's CZHub CI Team call, we decided that people prefer the Material theme and we should use that one.

**Pull request merged**